test/jest/Backup.spec.ts

```diff
@@ -0,0 +1,173 @@
+import fs from 'fs';
+import path from 'path';
+import Backup from '../../lib/backup/Backup';
+
+const removeDir = (dir: string) => {
+  if (fs.existsSync(dir)) {
+    fs.readdirSync(dir).forEach((file) => {
+      fs.unlinkSync(`${dir}/${file}`);
+    });
+  }
+
+  fs.rmdirSync(dir);
+};
+
+const wait = (ms: number) => {
+  return new Promise(resolve => setTimeout(resolve, ms));
+};
+
+const backupdir = 'backup-test';
+
+const raidenDatabase = 'raiden';
+const xudDatabase = 'xud';
+
+const backups = {
+  lnd: {
+    event: 'lnd event',
+    startup: 'lnd startup',
+  },
+  raiden: {
+    event: 'raiden event',
+    startup: 'raiden startup',
+  },
+  xud: {
+    event: 'xud event',
+    startup: 'xud startup',
+  },
+};
+
+let channelBackupCallback: any;
+
+const onListenerMock = jest.fn((event, callback) => {
+  expect(event).toEqual('channelBackup');
+
+  channelBackupCallback = callback;
+});
+
+jest.mock('../../lib/lndclient/LndClient', () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      on: onListenerMock,
+      currency: 'BTC',
+      init: () => Promise.resolve(),
+      close: () => Promise.resolve(),
+      subscribeChannelBackups: () => Promise.resolve(),
+      exportAllChannelBackup: () => Promise.resolve(backups.lnd.startup),
+    };
+  });
+});
+
+describe('Backup', () => {
+  const backup = new Backup();
+
+  beforeAll(async () => {
+    await Promise.all([
+      fs.promises.writeFile(
+        raidenDatabase,
+        backups.raiden.startup,
+      ),
+      fs.promises.writeFile(
+        xudDatabase,
+        backups.xud.startup,
+      ),
+    ]);
+
+    await backup.start({
+      backupdir,
+      loglevel: 'error',
+      dbpath: xudDatabase,
+      raiden: {
+        database: raidenDatabase,
+      },
+    });
+  });
+
+  test('should resolve backup directory', () => {
+    expect(backup['getDefaultBackupDir']()).toContainEqual('/');
+  });
+
+  test('should write LND backups on startup', () => {
+    expect(
+      fs.readFileSync(
+        path.join(backupdir, 'lnd-BTC'),
+        'utf-8',
```

Comment:
I'm guessing `utf-8` might just be an alias for `utf8` internally in nodejs such that it doesn't matter, but I noticed again that it'd different from what I'm used to. I won't comment on the other instances.